components/builder-db/src/data_store.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error;
+use std::convert::From;
+
+use pool::Pool;
+use migration::Migrator;
+use error::Error;
+
+pub trait DataStore<E: From<Error>> {
```

Comment:
This would be a great place to use an [associated type](https://doc.rust-lang.org/book/associated-types.html) instead of a standard generic for the `Error` type